screens/payment/TopUp.tsx

```diff
@@ -0,0 +1,157 @@
+      <SecondaryButton
+        title={leftText}
+        onPress={() => onButtonClicked(leftAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === leftAmount ? Colors.red[400] : Colors.gray[700],
+        }}
+      />
+      <SecondaryButton
+        title={rightText}
+        onPress={() => onButtonClicked(rightAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === rightAmount ? Colors.red[400] : Colors.gray[700],
+        }}
+      />
+    </View>
+  );
+};
+
+const TopUp: React.FC<TopUpProps> = ({ navigation, route }) => {
+  const [amount, setAmount] = useState<number>(0);
+  const [isManualChooseAmount, setIsManualChooseAmount] =
+    useState<boolean>(false);
+  const [activeButton, setActiveButton] = useState<number>();
+
+  const onPressButton = (amount: number) => {
+    setActiveButton(amount);
+    setAmount(amount);
+    setIsManualChooseAmount(amount === 0 ? true : false);
+  };
+
+  return (
+    <BodyContainer>
+      <SubHeaderText text="Current Balance" />
+      <View style={styles.currentBalance}>
+        <HeaderText text={`฿ ${route.params.balance}`} />
+      </View>
+      {isManualChooseAmount && (
+        <TextInput
+          title="Please enter amount (50 baht minimum)"
+          errorText={
+            Number(amount) < 50 ? "Please top up at least 50 baht." : ""
+          }
+          placeholder="0"
+          value={amount.toString()}
+          onChangeText={(amount) => setAmount(Number(amount))}
+          inputMode={InputType.Numeric}
+        />
+      )}
+      <DoubleButton
+        leftAmount={100}
+        rightAmount={250}
+        onButtonClicked={onPressButton}
+        activeButton={activeButton}
+      />
+      <DoubleButton
+        leftAmount={500}
+        rightAmount={750}
+        onButtonClicked={onPressButton}
+        activeButton={activeButton}
+      />
+      <DoubleButton
+        leftAmount={1000}
+        rightAmount={0}
+        onButtonClicked={onPressButton}
+        activeButton={activeButton}
+      />
+      <PrimaryButton
+        title="Next"
+        onPress={() => {
+          if (Number(amount) >= 50) {
+            navigation.navigate("PaymentStack", {
+              screen: "PaymentOptions",
+              params: { amount: amount },
+            });
+          }
+        }}
+      />
+    </BodyContainer>
+  );
+};
+
+export default TopUp;
+
+const styles = StyleSheet.create({
+  currentBalance: {
+    flexDirection: "row",
+    alignItems: "center",
+    backgroundColor: Colors.white,
+    borderRadius: 8,
+    padding: 8,
+    paddingVertical: Platform.OS === "android" ? 13 : 16,
+    shadowColor: Colors.black,
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.2,
+    shadowRadius: 2,
+    elevation: Platform.OS === "android" ? 4 : 2,
```

Comment:
No need to check android anymore since IOS doesn't use this

Reply:
Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)